packages/mobile/src/navigator/Navigator.tsx

```diff
@@ -475,14 +483,11 @@ export function MainStackScreen() {
 
     if (!language) {
       initialRoute = Screens.Language
-    } else if (!e164Number) {
+    } else if (!e164Number || !acceptedTerms || pincodeType === PincodeType.Unset) {
+      // User didn't go far enough in onboarding, start again from education
       initialRoute = Screens.OnboardingEducationScreen
-    } else if (!acceptedTerms) {
-      initialRoute = Screens.RegulatoryTerms
-    } else if (pincodeType === PincodeType.Unset) {
-      initialRoute = Screens.PincodeSet
     } else if (!redeemComplete && !account) {
-      initialRoute = Screens.EnterInviteCode
+      initialRoute = isRestoringAccount ? Screens.ImportWallet : Screens.EnterInviteCode
```

Comment:
I would have guessed that `isRestoringAccount` is for a progress indicator. Maybe `choseToRestoreAccount`?

Reply:
Yes, thanks for the suggestion.